_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp

```diff
@@ -2032,7 +2032,11 @@ mfxStatus CheckVideoParam(MfxVideoParam& par, ENCODE_CAPS_HEVC const & caps, boo
     changed += CheckMax(par.mfx.NumRefFrame, maxDPB - 1);
 
     if (par.mfx.NumRefFrame)
-        maxDPB = par.mfx.NumRefFrame + 1;
+    {
+        // If NumActiveRef parameters are set already, DPB size should not be less than NumActiveRef+1
+        maxDPB = std::max({ par.mfx.NumRefFrame + 1, par.m_ext.DDI.NumActiveRefP + 1,
+            par.m_ext.DDI.NumActiveRefBL0 + 1, par.m_ext.DDI.NumActiveRefBL1 + 1 });
```

Comment:
Shouldn't it be "par.m_ext.DDI.NumActiveRefBL0 + par.m_ext.DDI.NumActiveRefBL1 + 1" instead of
"par.m_ext.DDI.NumActiveRefBL0 + 1, par.m_ext.DDI.NumActiveRefBL1 + 1" ?

Reply:
Fixed